### DISCUSSION TITLE
Connessione HA da esterno con Duckdns

### DISCUSSION THREAD

**Discussion by FlavioRadice** [indomus/forum#2794] [HUB: Home Assistant]
Ciao, ho configurato HA per accedere dall'esterno tramite duckdns, ad un certo punto non ha più funzionato (dopo mesi ) non so se ho installato qualcosa che è entrato in conflitto, oppure un aggiornamento automativo.  Dal mio iphone se tento di collegarmi con HA tramite l'APP mi ritorna questo errore NSURLErrorDomain 1004

**Comment by snakuzzo** [comment]
Dalla documentazione Apple...

https://developer.apple.com/documentation/foundation/1508628-url_loading_system_error_codes/nsurlerrorcannotconnecttohost#discussion

...potrebbe dipendere da tante cose.
Hai verificato che il certificato non sia scaduto?
Fai delle prove da browser e non dall'app companion

**Reply by FlavioRadice** [reply]
ciao, grazie per la risposta, ho provato anche da browser e non capisco, perche http://homeassistantflavio.duckdns.org risponde al ping
Sul router avevo già messo le istruzioni per il routing alla porta 443
il certificato sembra non scaduto (scusa non sono super esperto, penso si possa vedere dal log)

s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
# INFO: Using main config file /data/workdir/config
+ Account already registered!
[14:25:08] WARNING: 
[14:25:19] INFO: Renew certificate for domains: homeassistantflavio.duckdns.org and aliases: 
# INFO: Using main config file /data/workdir/config
Processing homeassistantflavio.duckdns.org
 + Checking domain name(s) of existing cert... unchanged.
 + Checking expire date of existing cert...
 + Valid till Feb 29 16:30:34 2024 GMT (Longer than 30 days). Skipping renew!
[14:31:06] INFO: OK
188.218.16.149

NOCHANGE
[14:36:07] INFO: OK
188.218.16.149

**Comment by FlavioRadice** [comment]
Buongiorno,
scusami, ho smanettato tutta notte alla fine ho trovato su internet uno
schema sulle porte da settiare per il router, aggiungendo sia in ingresso
che in uscita il roting per la porta 8123 , adesso funziona,

Non comprendo come mai abbia funzionato per settimane senza questa
modifica...

Grazie per il tuo supporto

Il giorno mar 26 dic 2023 alle ore 15:04 Diego Ambrosanio <
***@***.***> ha scritto:

> Dalla documentazione Apple...
>
>
> https://developer.apple.com/documentation/foundation/1508628-url_loading_system_error_codes/nsurlerrorcannotconnecttohost#discussion
>
> ...potrebbe dipendere da tante cose.
> Hai verificato che il certificato non sia scaduto?
> Fai delle prove da browser e non dall'app companion
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/indomus/forum/discussions/2794#discussioncomment-7949177>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A44VELCYUXQHWQV6IW6MAKDYLLKPPAVCNFSM6AAAAABBDAPZIGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TSNBZGE3TO>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

*Flavio Radice*
✉️  ***@***.***
[image: 📞] +39 335 7266804 <+39%20335%20726%206804>

**Reply by snakuzzo** [reply]
Leggo solo ora.
Pro futuro:

1. Quando fai reply, fallo sotto l'ultimo commento e non creando nuovi thread come hai appena fatto
2. Non rispondere ai messaggi via mail (come hai fatto ora), ma accedendo a Discussions
3. Quando posti codice o log, usa il markdown. Così come hai fatto ora è uno schifo, e puoi notarlo stesso tu.

Tornando al tuo problema, cosa vorrebbe dire il routing sia in ingresso che in uscita ? L'unica cosa che occorre, per come hai configurato tu HA, è l'inoltro del traffico (Port forwarding) dalla porta esterna che scegli con protocollo TCP verso IP e porta interna su cui è in ascolto il servizio di Core...non occorre altro. Se hai dovuto fare configurazioni strane, vuol dire che chissà che casino hai fatto

**Reply by FlavioRadice** [reply]
ciao, scusa, cercherò di imparare ...
volevo dire che solo con l'aggiunta dell'ultima istruzione nel router, adesso duckdns funziona correttamente
![Immagine 2023-12-27 162512](https://github.com/indomus/forum/assets/121197100/4014a33b-e866-42c0-b054-6a30c0127c7a)

**Reply by snakuzzo** [reply]
Grazie al cavolo che funziona.
Se hai compreso a cosa serve il Port forwarding, quell'ultima regola consente di raggiungere HA da internet sulla porta 8123, il cui traffico verrà veicolato verso l'IP del tuo server sulla porta 8123 (porta di ascolto di HA). 
Detto ciò, le altre due non servono assolutamente a nulla. Quella con protocollo UDP è totalmente inutile e non voglio nemmeno sapere dove hai letto di doverla aprire. La prima invece ti permette di raggiungere HA sulla porta 443...ma devi deciderti...non ha senso raggiungere HA da internet su due porte diverse

**Reply by FlavioRadice** [reply]
lo strano è che è funzionato per 2 settimane solo con le prime 2 regole... scusa, ma ho vaghe reminiscenze informatiche ...

**Reply by snakuzzo** [reply]
Sì ma funzionava perché lo raggiungevi sulla porta 443 dall'esterno. Tu come vuoi raggiungere HA dall'esterno? Su quale porta?
